c2rust-analyze/src/context.rs

```diff
@@ -182,6 +193,27 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
             self.assign_pointer_to_field(field);
         }
     }
+
+    pub fn fn_failed(&mut self, did: DefId) -> bool {
+        self.fns_failed.contains_key(&did)
+    }
+
+    pub fn mark_fn_failed(&mut self, did: DefId, detail: PanicDetail) {
+        if self.fns_failed.contains_key(&did) {
+            return;
+        }
+
+        self.fns_failed.insert(did, detail);
+
+        // This is the first time marking `did` as failed, so also mark all of its callers.
+        let callers = self.fn_callers.get(&did).cloned().unwrap_or(Vec::new());
```

Comment:
```suggestion
        let callers = self.fn_callers.get(&did).cloned().unwrap_or_default();
```

Reply:
This only removes a type hint that might be useful for the reader.

Reply:
This is a `clippy::or_fun_call` warning.  I don't see a good reason not to follow it.  The fact that it's `Vec` isn't important here.

Reply:
This is a false positive in Clippy.  "This is only bad if it allocates or does some non-trivial amount of work," but "The lint also cannot figure out whether the function you call is actually expensive to call or not."  In this case the function being called is `const`, so it doesn't need to do any run-time work at all.  The lint in question has also been [downgraded to allow-by-default](https://github.com/rust-lang/rust-clippy/pull/9829) due to the high rate of false positives.

Reply:
This one is a slightly different warning.  When it suggests using a closure that calls a `const fn`, that is more of a false warning as you were saying, but in this case, `.unwrap_or_default()` is functionally different (I'm surprised it falls under the same `clippy` lint, as it's quite different).  Using `Default::default()` for this is more semantically clear (we don't care that it's actually `Vec::new()`, but that it's the default, empty container).  We could conceivably change it to `ThinVec`, `Box<[T]>`, `IndexSet`, etc., in which case the `.unwrap_or_default()` wouldn't need to be changed since it's the same semantic logic.

That said, I don't really want to spend time debating the merits of `clippy` lints.  I think we should just do what `clippy` says here.  There's no major reason not to, and using `Default` here is more semantically specific.

Reply:
>I think we should just do what clippy says here. There's no major reason not to

There's no good reason to make this change.  The fact that Clippy suggested it isn't a good reason on its own, because Clippy often gives bad advice.  (Any clippy lints that do regularly give good advice tend to get uplifted into rustc.)

> The fact that it's Vec isn't important here.

I disagree.  Knowing the type of `xs` is useful for knowing what you'll get if you do `for x in xs`.

As far as I know, the official policy for this repo does not require the code to be clippy-clean.  Unless that changes (which I would oppose), I don't see any value in making the code (very slightly) worse just to appease the tool.